contents/docs/product-analytics/best-practices.mdx

```diff
@@ -0,0 +1,104 @@
+---
+title: Product analytics best practices
+sidebar: Docs
+showTitle: true
+---
+
+## 1. Set up a reverse proxy
+
+Tracking and ad blockers can prevent your events from being sent to PostHog. By setting up a reverse proxy, you can send events to PostHog using your own domain. This means that events are less likely to be blocked, and so you'll be able to capture more data.
+
+See our docs on [how to set up a reverse proxy](/docs/advanced/proxy) for more details on how to do this.
+
+## 2. Implement a naming convention
+
+As your product grows, the amount of analytics data grows with it. Without naming conventions, your data quickly becomes difficult to work with.
+
+For example, let's say you're capturing signup metrics. Your team might capture events differently and with multiple names:
+
+- Web developer: Captures `Create Account` when a user clicks a button to begin signing up.
+- iOS developer: Captures the same event but names it `user_sign_up`.
+- Android developer: Calls it `create-account` but only captures it when they have finished signing up.
+
+All these differences make it more likely that you'll misinterpret your data. Implementing a naming convention prevents these problems.
+
+### Suggested naming guide
+
+- Only use lowercase letters.
+- Use present-tense verbs, e.g., "submit" and "create" instead of "submitted" and "created".
+- Use snake case, i.e., `signup_button_click`
+- Create a dedicated list of allowed verbs, and don't deviate from it. Here's a list of the most common ones you're likely to use:
+```
+- click
+- submit
+- create
+- view
+- add
+- invite
+- update
+- delete
+- remove
+- start
+- end
+- cancel
+- fail
+- generate
+- send
+```
+
+- For event names, use the `category:object_action` framework:
+
+  - `category` describes the context the event took place in – e.g., `account_settings` or `signup_flow` 
+
+  - `object` is a noun that refers to the component or location the event took place in, like `forget_password_button` or `pricing_page`.
+
+  - `action` is a verb that describes what happened, like `click`, `submit`, or `create`
+
+  - Putting all of these together, you get `account_settings:forget_password_button_click` or `signup_flow:pricing_page_view` – The order is important as it makes it easier to find a specific event within a long list.
+
+- For property names:
+
+  - Use the `object_adjective` pattern. For instance: `user_id`, `item_price`, or `member_count`.
+
+  - Use `is/has` prefixes for boolean properties. For example: `is_subscribed`, `has_seen_upsell`.
+
+  - If the property's value is a date or timestamp, include _date or _timestamp at the end. For example, `user_creation_date`, `last_login_timestamp`.
+
+## 3. Version your events
```

Comment:
Does anyone actually do this? There is no easy way to know which version you are on.

I think a better suggestion might be to verify or add description to your events in PostHog. 

Like this: https://us.posthog.com/project/2/data-management/events/01790132-e887-0000-ce39-e2b1340a62c4/edit

Reply:
People do this and its quite useful. Descriptions are good too, but harder to understand at a glance e.g., when selecting an event name from an insight, you need to hover over the event name. 
also, event names are added in code, but descriptions in PostHog, so require an extra place to do things